src/com/owncloud/android/ui/adapter/LocalFileListAdapter.java

```diff
@@ -104,6 +107,12 @@ public View getView(int position, View convertView, ViewGroup parent) {
             fileName.setText(name);
             
             ImageView fileIcon = (ImageView) view.findViewById(R.id.thumbnail);
+
+            /** Cancellation needs do be checked and done before changing the drawable in fileIcon, or
+             * {@link ThumbnailsCacheManager#cancelPotentialWork} will NEVER cancel any task.
+             **/
+            boolean allowedToCreateNewThumbnail = (ThumbnailsCacheManager.cancelPotentialWork(file, fileIcon));
```

Comment:
The method ```cancelPotentialWork(...)``` is not just generating the flag, but checking if the imageView 'fileIcon' has a ```ThumbnailGenerationTask``` in progress associated to its contained ```Drawable```, and *cancelling* it in case that it's loading the thumbnail for a file different than the current file.

Case: a thumbnail is started to be generated for a PNG file, the task is associated to the imageView for the thumbnail in the list; the user scrolls up before the thumbnail is ready; the ListAdapter resuses the view of the PNG file for a folder. With this line here, the task is cancelled ; with the line in the !directory branch, the tasks keeps running until saving the thumbnail in the cache. Not functionallly wrong, but with tons of images in the same folder let all the thumbnail tasks finishing results in a performnace problem.

There are other things that could be done to control better the performance in these cases, of course. Consider this patch a minimum effort to control the specific issue pointed, with the target of releasing ASAP.